src/hank_rule.erl

```diff
@@ -46,9 +46,6 @@ analyze(Rule, ASTs, Context) ->
 %% @doc Check if given rule should be ignored from results
 -spec is_ignored(t(), ignore_pattern(), [all | term()]) -> boolean().
 is_ignored(Rule, Pattern, IgnoredSpecs) ->
-    lists:any(fun(IgnoreSpec) ->
-                 IgnoreSpec =:= all
-                 orelse IgnoreSpec =:= Pattern
-                 orelse Rule:ignored(Pattern, IgnoreSpec)
+    lists:any(fun(IgnoreSpec) -> IgnoreSpec =:= all orelse Rule:ignored(Pattern, IgnoreSpec)
```

Comment:
I changed this to make `ignore/2` implementation more _intuitive_ (i.e. it's not _just the special cases_, it handles all scenarios).
The alternative would've been to make it an optional callback. Let me know if you prefer that.